src/providers/SpeechProvider/SpeechProvider.actions.js

```diff
@@ -124,15 +124,40 @@ export function changeRate(rate) {
 export function getVoices() {
   return dispatch => {
     dispatch(requestVoices());
-
     return tts.getVoices().then(pvoices => {
       let voices = [];
       try {
-        voices = pvoices.map(({ voiceURI, lang, name }) => ({
-          voiceURI,
-          lang,
-          name
-        }));
+        voices = pvoices.map(
+          ({
+            voiceURI,
+            lang,
+            name,
+            Locale,
+            ShortName,
+            DisplayName,
+            Gender
+          }) => {
+            let voice = {};
```

Comment:
It might be easier to do this transformation in `getVoices` (or maybe even `getAzureVoices`) since we already know if the voice is provided by Azure or not, and we wouldn't have to check if different properties are defined or not. We could also write some JSDoc for `getVoices` and have a custom `Voice` type:

```js
/**
 * @typedef {Object} Voice
 *
 * @description A voice that can be used for text-to-speech.
 *
 * @property {string} lang
 * The voice's language code. For example, "en-US".
 *
 * @property {string} voiceURI
 * The voice's identifier. For example, "en-US-AriaNeural".
 *
 * @property {('local'|'cloud')} voiceSource
 * Whether the voice is cloud-based and requires an Internet connection,
 * or is provided by the platform.
 *
 * @property {string} name
 * The voice's display name. For example, "Aria (en-US) - Female".
 */

/**
 * @returns {Promise<Voice[]>}
 */
async getVoices() {
  // ...
}
```

Reply:
great feedback, will implement it in the next update